### PR TITLE
Introduce Page.getSingleValuePage

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/TopNRowNumberOperator.java [hunk 1]
```diff
@@ -381,12 +381,7 @@ public boolean isEmpty()
 
     private static Block[] getSingleValueBlocks(Page page, int position)
     {
-        Block[] blocks = page.getBlocks();
-        Block[] row = new Block[blocks.length];
-        for (int i = 0; i < blocks.length; i++) {
-            row[i] = blocks[i].getSingleValueBlock(position);
-        }
-        return row;
+        return page.getSingleValuePage(position).getBlocks();
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/TopNRowNumberOperator.java]: Inline this method now?

File: presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java [hunk 2]
```diff
@@ -649,8 +649,7 @@ public IntPredicate getSpillMask()
 
         public SavedRow(Page page, int position, long joinPositionWithinPartition, boolean currentProbePositionProducedRow, int joinSourcePositions)
         {
-            this.row = page.mask(new int[] {position});
-            this.row.compact();
+            this.row = page.getSingleValuePage(position);
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java]: Can you double check that `mask` + `compact` has the same result as `getSingleValuePage`... specifically that we always end up with a copy.

Comment (findepi) [reply presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java]: `Page.compact` has `DictionaryBlock`-related logic which I don't understand, so I won't speak about this part.

Other than that, I looked through implementations of `Block. getSingleValueBlock` and everywhere it looks like simplified implementation of `Block.copyRegion`. There is always a copy.

**Comment by losipiuk** [comment]
Merged, thanks!

**Pull request merged**